torchao/prototype/float8nocompile/float8nocompile_linear.py

```diff
@@ -19,4 +36,149 @@ class Float8LinearNoCompile(torch.nn.Linear):
     Note: this is **prototype** and not suitable for production use.
     """
 
-    pass
+    def __init__(self, *args, **kwargs):
+        """
+        Additional arguments on top of `torch.nn.Linear`'s arguments:
+        * `config`: Float8LinearConfig
+        """
+
+        # Amax scales should always be kept as float32.
+        self.always_float32_buffers = set()
+        config = kwargs.pop("config")
+        emulate = config.emulate
+        super().__init__(*args, **kwargs)
+
+        # Defines the scaling behavior of input, weight, grad_output
+        self.scaling_type_input = config.cast_config_input.scaling_type
+        self.scaling_type_weight = config.cast_config_weight.scaling_type
+        self.scaling_type_grad_output = config.cast_config_grad_output.scaling_type
+
+        self.config = config
+        self.is_amax_initialized = not self.config.enable_amax_init
+
+        self.linear_mm_config = LinearMMConfig(
+            # output
+            ScaledMMConfig(
+                emulate,
+                self.config.gemm_config_output.use_fast_accum,
+                False,
+                self.config.pad_inner_dim,
+            ),
+            # grad_input
+            ScaledMMConfig(
+                emulate,
+                self.config.gemm_config_grad_input.use_fast_accum,
+                False,
+                self.config.pad_inner_dim,
+            ),
+            # grad_weight
+            ScaledMMConfig(
+                emulate,
+                self.config.gemm_config_grad_weight.use_fast_accum,
+                False,
+                self.config.pad_inner_dim,
+            ),
+        )
+
+    def forward(self, input: torch.Tensor) -> torch.Tensor:
+        # TODO(danielvegamyhre): modify to support for FSDP once dependencies are implemented
+        output = self.forward_fp8_matmul(input)
+        if self.bias is not None:
```

Comment:
feel free to not support bias at all to simplify, as our target model uses linear without bias